### ISSUE TITLE
Error: EPERM: operation not permitted, scandir

### ISSUE THREAD

**Issue by doberkofler** [npm/npm#17747]
#### I'm opening this issue because:

  - [x] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

```
D:\MyDev\ljs_app\trunk\periscope\build>npm i
npm ERR! path D:\MyDev\ljs_app\trunk\periscope\build\node_modules\fsevents\node_modules
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall scandir
npm ERR! Error: EPERM: operation not permitted, scandir 'D:\MyDev\ljs_app\trunk\periscope\build\node_modules\fsevents\node_modules'
npm ERR!  { Error: EPERM: operation not permitted, scandir 'D:\MyDev\ljs_app\trunk\periscope\build\node_modules\fsevents\node_modules'
npm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'D:\\MyDev\\ljs_app\\trunk\\periscope\\build\\node_modules\\fsevents\\node_modules\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'scandir',
npm ERR!   path: 'D:\\MyDev\\ljs_app\\trunk\\periscope\\build\\node_modules\\fsevents\\node_modules' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\doberkofler.LBITS\AppData\Roaming\npm-cache\_logs\2017-07-12T18_53_54_304Z-debug.log
```

#### How can the CLI team reproduce the problem?

https://gist.github.com/doberkofler/f24eea4a2d57cefde864ed64556b7bc1.js

### supporting information:

 - `npm -v` prints: 5.2.0
 - `node -v` prints: 8.1.4
 - `npm config get registry` prints: http://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Windows
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by jdrake3** [comment]
Me too, but `node -v` prints: **v6.10.0**

**Comment by doberkofler** [comment]
Unfortunately this problem is so nasty, it forced me to revert back to npm 4. I loved the performance of version 5 and would really hope to get this fixed. Thank you!

**Comment by marcins** [comment]
This seems to be a duplicate of #17671, or at least has the same symptoms. That issue has more detail though.

**Comment by doberkofler** [comment]
Tested again with:

* npm -v prints: 5.3.0
* node -v prints: 8.3.0

but the problem persists and had to move back to npm 4!

**Comment by doberkofler** [comment]
@marcins I'm not so sure if this is really the same problem...

**Comment by doberkofler** [comment]
I have some news that might (hopefully) improve the chances to track this problem down.
I discovered that after cleaning the cache the first installation works as expected but when later installing the same dependencies in another project the error arises. In my opinion and given the fact that the problems seems to happen with optional dependencies, this suggests a problem in the resolution/caching of the resolved optional dependencies. I hope this helps and hope even more to have this fixed and finally being able to use the latest (and greatest) npm version.

**Comment by RGTGT** [comment]
Same issue here:
- npm -v prints: 5.3.0
- node -v prints: v6.11.1

I'm on windows

Although, upgrading to node v6.11.2 fixed my errors.

**Comment by srsgores** [comment]
I get this too, on Windows.  Even on the latest `8.4.0`.  I'll see if downgrading to LTS `6.11.2` fixes this.

**Comment by srsgores** [comment]
I still get the same problem on LTS `6.11.2` (so this error prevents me from using Gatsby entirely on WIndows):

> npm ERR! Windows_NT 10.0.15063
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "gatsby"
npm ERR! node v6.11.2
npm ERR! npm  v3.10.10
npm ERR! path C:\Users\user\github\project\node_modules\gatsby\bin\gatsby.js
npm ERR! code ENOENT
npm ERR! errno -4058
npm ERR! syscall chmod
npm ERR! enoent ENOENT: no such file or directory, chmod 'C:\Users\user\github\project\node_modules\gatsby\bin\gatsby.js'
npm ERR! enoent ENOENT: no such file or directory, chmod 'C:\Users\user\github\project\node_modules\gatsby\bin\gatsby.js'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent
npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\user\github\project\npm-debug.log

**Comment by marcins** [comment]
@srsgores I don't think that's the same issue, this issue relates to an `EPERM` operation not permitted error (which usually occurs intermittently when installing `fsevents`).

It looks like in this case the `gatsby.js` binary is assuming the `chmod` command exists, which is a Unix command and doesn't exist on Windows.

**Comment by srsgores** [comment]
@marcins, I definitely **do** get the error:

> npm WARN Error: EPERM: operation not permitted, scandir 'C:\Users\user\github\project\node_modules\find-up\node_modules'
npm WARN  { Error: EPERM: operation not permitted, scandir 'C:\Users\user\github\project\node_modules\find-up\node_modules'
npm WARN   stack: 'Error: EPERM: operation not permitted, scandir \'C:\\Users\\user\\github\\project\\node_modules\\find-up\\node_modules\'',
npm WARN   errno: -4048,
npm WARN   code: 'EPERM',
npm WARN   syscall: 'scandir',
npm WARN   path: 'C:\\Users\\user\\github\\project\\node_modules\\find-up\\node_modules' }

**Comment by haldhafari82** [comment]
if using cygwin:

 runas root

**Comment by akshayefusion** [comment]
Facing the same issue 
npm ERR! path D:\xampp\htdocs\east-hem\front\node_modules\fsevents\node_modules\
ansi-regex\package.json
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall unlink
npm ERR! Error: EPERM: operation not permitted, unlink 'D:\xampp\htdocs\east-hem
\front\node_modules\fsevents\node_modules\ansi-regex\package.json'
npm ERR!  { Error: EPERM: operation not permitted, unlink 'D:\xampp\htdocs\east-
hem\front\node_modules\fsevents\node_modules\ansi-regex\package.json'
npm ERR!   stack: 'Error: EPERM: operation not permitted, unlink \'D:\\xampp\\ht
docs\\east-hem\\front\\node_modules\\fsevents\\node_modules\\ansi-regex\\package
.json\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'unlink',
npm ERR!   path: 'D:\\xampp\\htdocs\\east-hem\\front\\node_modules\\fsevents\\no
de_modules\\ansi-regex\\package.json' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

**Comment by adamzhu-xs** [comment]
Got same issue. Seems NPM try to scan folder \test-app\node_modules\fsevents\node_modules which doesn't exist.

windows 7
node v6.10.3
npm 5.3.0

1. If i keep trying to install package, 1 out of 5 installing will work.
2. If i clean NPM cache first by force, next npm install will work.

**Comment by AleCaste** [comment]
I was getting the same error on Windows 7 and npm 5.3.0.
I ran ```npm cache verify``` and after that I had no problems installing modules.
I decided not to try ```npm cache clean --force``` since I did not want to empty the cache.

Give it a go to check if it helps.

**Comment by rossbu** [comment]
Same error with below Env:
    Win 7 Pro
    Node 8.2.1
    NPM 5.3.X
    vscode terminal with git bash. 
    ( Note: be alert how you open vscode and terminal and what user do you  use to open it)

Solution: ( with VS Code terminal +  git bash ) 
1. remove node_modules
2. npm cache verify
3. npm install
4. ng serve.

Error is gone.   good luck.

**Comment by billysean** [comment]
having this issue as well.. NPM 5.3, node 8.4
tried cleaning cache, verifying cache as suggested but problem still preserve..
on second thought, my problem is gone when using windows powershell admin

**Comment by raucuqua** [comment]
work very fine
npm 5.3 + node 6

**Comment by adriatic** [comment]
I am fighting this for more than a year, so clearly this is independent of npm and or node version. Just now, I got:
```
λ npm install
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
npm WARN deprecated express@2.5.11: express 2.x series is deprecated
npm WARN deprecated connect@1.9.2: connect 1.x series is deprecated
npm ERR! path H:\work\aurelia-tools\acb-samples\skeleton-typescript-cli\node_modules\fsevents\node_modules\getpass\node_modules
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall scandir
npm ERR! Error: EPERM: operation not permitted, scandir 'H:\work\aurelia-tools\acb-samples\skeleton-typescript-cli\node_modules\fsevents\node_modules\getpass\node_modules'
npm ERR!  { Error: EPERM: operation not permitted, scandir 'H:\work\aurelia-tools\acb-samples\skeleton-typescript-cli\node_modules\fsevents\node_modules\getpass\node_modules'
npm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'H:\\work\\aurelia-tools\\acb-samples\\skeleton-typescript-cli\\node_modules\\fsevents\\node_modules\\getpass\\node_modules\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'scandir',
npm ERR!   path: 'H:\\work\\aurelia-tools\\acb-samples\\skeleton-typescript-cli\\node_modules\\fsevents\\node_modules\\getpass\\node_modules' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\nik\AppData\Roaming\npm-cache\_logs\2017-09-15T21_38_03_356Z-debug.log
```
Then, I just removed the node_modules and run `nom install` again:
```
npm install
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
npm WARN deprecated express@2.5.11: express 2.x series is deprecated
npm WARN deprecated connect@1.9.2: connect 1.x series is deprecated
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 1042 packages in 69.35s
```

I am pretty sure that there is more than one source of this same manifestation, that include the "interference" from running instance of vscode as well as various virus and malware checkers (I have both of them running - and at this time, vscode was not).

The problem is obviously npm's failure to execute:

```
scandir 'H:\work\aurelia-tools\acb-samples\skeleton-typescript-cli\node_modules\fsevents\node_modules\getpass\node_modules'
```

because "some other running process* has a lock on that folder. This problem is further compounded by the fact that this folder is temporary, making it impossible to look for the lock owner after the fact. Probably the best approach would be that npm "learns" some of the tricks explained in https://superuser.com/questions/117902/find-out-which-process-is-locking-a-file-or-folder-in-windows article and dump that data into its log.

**Comment by marcins** [comment]
@adriatic is this with npm 5.4.2? That version (just released) has a fix for this issue, so would be good to know If it isn’t fixed.

**Comment by akshay-js** [comment]
After downgrade npm version to 4.6, now working

**Comment by doberkofler** [comment]
@marcins I've tried with the latest npm 5.4.2 version and node 8.5.0 but the error persists and I one again had to move back to npm 4.

**Comment by corvinrok** [comment]
Is there a problem with downgrading to 4.6 when you are running node 8.5.0? I have the same problem listed in this thread with no version 5.3.x or 5.4.x working , but downgrading back to npm 4.x was giving me other errors before and I didn't know if node 8.5.0 has some sort of minimum version requirement on the npm version it would work with....

**Comment by corvinrok** [comment]
I can confirm that downgrading to the specific version of npm 4.6.1 did work for me, running a node 8.5.0 project.

**Comment by ceriath** [comment]
@marcins With 5.4.2 you can run any npm install/update (optionally -g). But only once, after that the error still occurs. Running npm cache verify fixes it again for one run, so basically after every npm install run npm cache verify as a workaround.

A fix would be nice tho

**Comment by cjablonski76** [comment]
@marcins I tried on npm@5.4.2 and the first `npm install` succeeded. If I run it again it fails. If I continue to run `npm install` I see it succeed once in awhile and fail other times.

**Comment by scoop2** [comment]
same issue,
win7-64
npm v7.9.0

-npm cache verify
solved it

**Comment by acsakshay** [comment]
As previously suggested I bypassed the problem by downgrading to a much lower 5.x version. I experienced this issue just after upgrading npm to the recent version 5.4.2. Can confirm even downgrading to 5.3.0 doesn't help

`npm install npm@5.0.1 -g`

**Comment by gkorland** [comment]
Any progress on this issue? we managed to reproduce the issue on npm 5.3.0, 5.4.0 & 5.4.2.

**Comment by grbspltt** [comment]
I was having the same issue on a new machine.  I turned UAC down to the lowest setting and no longer have a problem.

**Comment by doberkofler** [comment]
@zkat @marcin-wosinek Is there any update on this issue? It has been a while and windows users basically are stuck with v4.

**Comment by steffanhalv** [comment]
I have this problem to..

**Comment by doberkofler** [comment]
@zkat @marcin-wosinek Just tried with the latest version 5.5.1 and still unable to use v5 on the Windows platform. Had again to move back to v4!

```
D:\MyDev\ljs_app\trunk\periscope\examples\webpack\basic>npm cache clean --force
npm WARN using --force I sure hope you know what you are doing.

D:\MyDev\ljs_app\trunk\periscope\examples\webpack\basic>npm cache verify
Cache verified and compressed (~\AppData\Roaming\npm-cache\_cacache):
Content verified: 0 (0 bytes)
Index entries: 0
Finished in 0.026s

D:\MyDev\ljs_app\trunk\periscope\examples\webpack\basic>npm i

> uglifyjs-webpack-plugin@0.4.6 postinstall D:\MyDev\ljs_app\trunk\periscope\examples\webpack\basic\node_modules\webpack\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js

added 403 packages in 26.097s

D:\MyDev\ljs_app\trunk\periscope\examples\webpack\basic>npm i
npm ERR! path D:\MyDev\ljs_app\trunk\periscope\examples\webpack\basic\node_modules\fsevents\node_modules\dashdash\node_modules
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall scandir
npm ERR! Error: EPERM: operation not permitted, scandir 'D:\MyDev\ljs_app\trunk\periscope\examples\webpack\basic\node_modules\fsevents\node_modules\dashdash\nod
e_modules'
npm ERR!  { Error: EPERM: operation not permitted, scandir 'D:\MyDev\ljs_app\trunk\periscope\examples\webpack\basic\node_modules\fsevents\node_modules\dashdash\
node_modules'
npm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\webpack\\basic\\node_modules\\fsevents\\node
_modules\\dashdash\\node_modules\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'scandir',
npm ERR!   path: 'D:\\MyDev\\ljs_app\\trunk\\periscope\\examples\\webpack\\basic\\node_modules\\fsevents\\node_modules\\dashdash\\node_modules' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\doberkofler.LBITS\AppData\Roaming\npm-cache\_logs\2017-10-16T05_08_02_492Z-debug.log

```

**Comment by doberkofler** [comment]
@grbspltt I've also tried with the lowest UAC setting several times but it makes no difference for me.

**Comment by ewhitmore** [comment]
We have tried the following without resolution:

- downgrading to 5.4
- downgrading to 5.04
- Setting UAC to lowest 
- npm cache verify
- tried installing different versions of node (LTS to current)

System:
Windows Server 2012 R2 - Patched as of 10/10/17

**Comment by johnmckay-reward** [comment]
I'm having this issue as well! Any update?

**Comment by doberkofler** [comment]
@zkat @iarna @marcins marcins An update on this would be really appreciated! It has been a while and a lot of windows users are still stuck with v4.

**Comment by marcin-wosinek** [comment]
I think you guys mean some other Marcin:)

**Comment by doberkofler** [comment]
@marcin-wosinek yes, sorry!

**Comment by shinjigi** [comment]
I solved partially by adding the parameter "**_--verbose_**" on every **install** process. It is strange but it works. Maybe it is a problem of sync with FS that is solved by overload the log process which gives more time to low-level operations....

**Comment by marcins** [comment]
@doberkofler I think maybe I was the Marcin you were looking for.

When I was investigating the issue I did make some notes about the root cause, see this comment: https://github.com/npm/npm/issues/17671#issuecomment-328083157

The key part is:

> Now the root cause of the issue on Windows appears to be that npm starts rolling back the install of `fsevents` before the `updatePackageJson` for it's children has completed. Since `updatePackageJson` is trying to write to files/directories that are in process of being deleted you end up with permission errors as while `rimraf` is trying to delete files they are being written to.

The change I made, forcing fsync on Windows, obviously didn't solve the root cause but similar to what @shinjigi mentioned might've just slowed down things enough that it doesn't always happen.  To fix the root cause would require much more significant re-architecture of how install / rollback works to avoid the conflicting tasks happening in parallel.

Unfortunately I don't have the time to work on this - I was doing it in my spare time, I don't use Windows day to day, but one person on my team does, so I have some vested interest but not enough. It requires a much deeper understanding of the npm internals than I managed to get in my initial investigation.

**Comment by steffanhalv** [comment]
I could install by running

npm install --force

👍😅

**Comment by doberkofler** [comment]
@marcins Thank you for your feedback and all help on this issue!
I've copied you because it seems as if this problem does not get the attention it should considering the number of people affected. Would you know of any way to escalate this and get some feedback from the core team?

**Comment by jdrake3** [comment]
I really hope this is some Halloween joke...

**Comment by doberkofler** [comment]
@marcins @zkat Instead of getting any feedback from the core team, the issue was now just closed. very disappointing!

**Comment by AleCaste** [comment]
This issue is affecting all Windows users with npm versions from 4.6 onward and they simply close it "as it has gone three days without activity"... Really???? So from their perspective this means that the issue has magically fixed itself.
To my humble opinion this is actually one of the MAJOR issues with NPM.

**Comment by electrotype** [comment]
It's not even hypothetical: we *are* going to try Yarn where I work, (mainly) because of this issue. Closing it is plain ridiculous.

**Comment by simonneedham** [comment]
I think we've reached the point where the conclusion has to be that npm does not support Windows.

**Comment by adriatic** [comment]
I agree with all comments made since this issue is closed. Using yarn instead of npm is the only proper response, which will make npm rethink their decision if we all do that

This problem is not really difficult to fix, since it is always caused by accessing a locked folder - either by a process created by npm (or its children) or some unrelated process owned by a virus checker of some kind. I am also assuming that there are many victims of this bug that would be very happy to help npm by running an instrumented version of npm that would help isolating the cause.

A lot stronger enforced compliance to version range for all components in npm registry should also help a lot. At the moment all these “please update your component” are not only annoying, but likely to contribute to problems.

**Comment by doberkofler** [comment]
Quite frustrating but let's try for the 3rd time to get any feedback: https://github.com/npm/npm/issues/19004

**Comment by LantosIstvan** [comment]
This one solved the #17671 error for me:

```sh
$ npm i -f <package>
# or
$ npm install --force <package>
```

Maybe related here. `fsevents` has missing dependencies without forced install since `v5.0.4`. Check the log files if it's complaining about missing `fsevents` dependencies and check it's folder also.

**Comment by corvinrok** [comment]
This issue has now become the main problem with node since 7.10.1.

Because of this, our project can't move past node 7.10.1, even though we would have liked to move to all latest versions of node 8 (and later to node 9..).. 

Has NPM decided that they will not fix this issue? It's hard to believe since anyone using Windows and installing a native dependency (from node-sass to xml libraries and more) will run into this issue.

We are considering moving to yarn because of this since it has killed our use of npm.

**Comment by Callie-C-Wang** [comment]
Hi @DJviolin 
Your solution did not work for me.
I can install the @angular/cli with -f option
However, when I try to create new application by command  ng new projName, it threw out error.

Here is the error message when I trying to install @angular/cli without -f option
> node-sass@4.5.3 install C:\Users\cwang\AppData\Roaming\npm\node_modules\@angular\cli\node_modules\node-sass
> node scripts/install.js


> uglifyjs-webpack-plugin@0.4.6 postinstall C:\Users\cwang\AppData\Roaming\npm\node_modules\@angular\cli\node_modules\webpack\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js

npm WARN Error: EPERM: operation not permitted, scandir 'C:\Users\cwang\AppData\Roaming\npm\node_modules\@angular\cli\node_modules'
npm WARN  { Error: EPERM: operation not permitted, scandir 'C:\Users\cwang\AppData\Roaming\npm\node_modules\@angular\cli\node_modules'
npm WARN   stack: 'Error: EPERM: operation not permitted, scandir \'C:\\Users\\cwang\\AppData\\Roaming\\npm\\node_modules\\@angular\\cli\\node_modules\'',
npm WARN   errno: -4048,
npm WARN   code: 'EPERM',
npm WARN   syscall: 'scandir',
npm WARN   path: 'C:\\Users\\cwang\\AppData\\Roaming\\npm\\node_modules\\@angular\\cli\\node_modules' }
npm ERR! file C:\windows\system32\cmd.exe;
npm ERR! path C:\windows\system32\cmd.exe;
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn C:\windows\system32\cmd.exe;
npm ERR! uglifyjs-webpack-plugin@0.4.6 postinstall: `node lib/post_install.js`
npm ERR! spawn C:\windows\system32\cmd.exe; ENOENT
npm ERR!
npm ERR! Failed at the uglifyjs-webpack-plugin@0.4.6 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

Any one know how to solve this problem?
Thank you.

**Comment by steffanhalv** [comment]
@corvinrok It was opened a new issue on this #19004 which is open and @KenanY added support label and windows label to this and the opened issue (which is not closed..). None still asigned thought

**Comment by corvinrok** [comment]
@steffanhalv thank you.

**Comment by warrenminano** [comment]
Successful installed using

`npm cache verify`
`npm install --force <package>`

npm -v prints: 5.5.1
node -v prints: 8.5.0

**Comment by OkanaByte** [comment]
npm ERR! path C:\xampp\htdocs\booksnotes\node_modules\fsevents\node_modules\dashdash\node_modules
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall scandir
npm ERR! Error: EPERM: operation not permitted, scandir 'C:\xampp\htdocs\booksnotes\node_modules\fsevents\node_modules\dashdash\node_modules'
npm ERR!  { Error: EPERM: operation not permitted, scandir 'C:\xampp\htdocs\booksnotes\node_modules\fsevents\node_modules\dashdash\node_modules'
npm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'C:\\xampp\\htdocs\\booksnotes\\node_modules\\fsevents\\node_modules\\dashdash\\node_modules\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'scandir',
npm ERR!   path: 'C:\\xampp\\htdocs\\booksnotes\\node_modules\\fsevents\\node_modules\\dashdash\\node_modules' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Wolf\AppData\Roaming\npm-cache\_logs\2017-11-05T23_53_40_753Z-debug.log

Package install failed, see above.

**Comment by matthiku** [comment]
node v8.9.0
npm v5.5.1
were just installed on Windows 10. I am using Git Bash.
Installed **vue-cli,** 
then created a new project use _vue-init webpack-simple stocktrader_, 
then did _npm install_ 
and _npm run dev_.
I broke out of the dev server (hit Ctrl-C twice!) and tried to install vuex: 
_npm install --save-dev vuex_ and got this error:
```
npm ERR! path C:\laragon\www\stockTrader\node_modules\fsevents\node_modules
npm ERR! code EPERM
npm ERR! errno -4048
npm ERR! syscall scandir
npm ERR! Error: EPERM: operation not permitted, scandir 'C:\laragon\www\stockTrader\node_modules\fsevents\node_modules'
npm ERR!  { Error: EPERM: operation not permitted, scandir 'C:\laragon\www\stockTrader\node_modules\fsevents\node_modules'
npm ERR!   stack: 'Error: EPERM: operation not permitted, scandir \'C:\\laragon\\www\\stockTrader\\node_modules\\fsevents\\node_modules\'',
npm ERR!   errno: -4048,
npm ERR!   code: 'EPERM',
npm ERR!   syscall: 'scandir',
npm ERR!   path: 'C:\\laragon\\www\\stockTrader\\node_modules\\fsevents\\node_modules' }
npm ERR!
npm ERR! Please try running this command again as root/Administrator.
```

Then I did a **npm cache verify** and tried to **install vuex again**, and bingo! it worked.

**Comment by ewhitmore** [comment]
FYI - this issue has been reposted here:  https://github.com/npm/npm/issues/19004 as this issue was closed by the bot.  Please comment there so the bot doesn't close that ticket as well.

**Comment by vahid8261** [comment]
I ran it with node.js command prompt and got exactly the same problem and when i change to windows command prompt it is fixed .

**Comment by puttuo** [comment]
In Windows, start the command line using "Run as administrator" seems can solve this issue.

**Comment by ewertonrfd** [comment]
For me, every time I run a command it gives an error. Then the second time it works out.
Ex: npm i --save-dev webpack@1.14.0 webpack-dev-server@1.16.2

1 st wrong
Right 2

I use windows 7 x64
npm --v 5.5.1
CMD: cmder

**Comment by mrTurkay** [comment]
I solved that issue with **yarn**

then you need to set default ng promt as yarn

**Comment by mseltene** [comment]
This worked for me: 
npm install scandir -g

and then continue, such as npm install.

**Comment by ewertonrfd** [comment]
@micsel    The command works
Thanks for the help, I've done 2 tests here ... I hope the problem does not come back

**Comment by IzikM** [comment]
Beautifull solution. Worked here, too.
After executing "**_npm install scandir -g_**" like @micsel suggested.
Thanks @micsel.

**Comment by micsel** [comment]
welcome! I'm glad I shared to help you all.

**Comment by IzikM** [comment]
Correction. After it worked - the error reared its head up again.
Following @micsel's example I had to run as an admin and execute:
"**npm cache verify**"
and then:
"**_npm install scandir rmdir fsync --save_**"
This worked. 
Hoping it sticks :-)

**Comment by matthiku** [comment]
I think this whole issue is just an intermittent problem related to some sync issues between Git bash and Windows. 
If the error happens, just repeat your last command and chances are high that it works then. 
You might also want to run a `npm cache verify` in between to help matters settle.

**Comment by tomcon** [comment]
@shinjigi solution worked for me:

> I solved partially by adding the parameter "--verbose" on every install process. It is strange but it works.

**Comment by ajochems** [comment]
In my case it was caused by Windows Defender realtime protection on Windows 10. As soon as i added an exception for the node.exe process all errors where gone!

**Comment by matthiku** [comment]
**Repeat the last command and problem is solved:**

![clipboarder 2017 12 01](https://user-images.githubusercontent.com/12457522/33480562-f1f611a4-d688-11e7-829b-01d8426b544d.png)

And it's always this **annoying 'fsevent' package** that causes it!!

**Comment by jerronvwh** [comment]
worked after 
npm install npm@5.6 -g

**Comment by bulbul16** [comment]
I was facing the same issue :What I have done is clear cashe of npm and its working for me. Please do the following
1. Uninstall node
2. Run the following command %programfiles%\nodejs and delete files
3. Run %appdata%\npm and delete files
4. Run %appdata%\npm-cache and delete files
5. Install node again whatever version you are willing to
6. Install angular cli again with this command
npm install -g angular-cli
Now try ng new app
It's working for me. Happy coding :)

**Comment by hsingh419** [comment]
upgrading with npm install npm@5.6 -g worked for me

Thanks!

**Comment by JimEastburn** [comment]
npm install npm@5.6 -g worked for me

**Comment by marcelovue** [comment]
Windows 10, I did run the CMD as root and I type 
`npm install babel-core@6` and works fine.
Without CMD as root I got that error related above

**Comment by zkat** [comment]
This seems fixed, so I'm gonna go ahead and lock it, since it's already closed.
